packages/mobile/views/dashboard/drawers/activity/views/ActivityDetails.svelte

```diff
@@ -0,0 +1,111 @@
+<script lang="typescript">
+    import { localize } from '@core/i18n'
+    import { isStrongholdUnlocked } from '@core/profile-manager'
+    import {
+        Activity,
+        ActivityAsyncStatus,
+        ActivityDirection,
+        ActivityType,
+        claimActivity,
+        getAssetFromPersistedAssets,
+    } from '@core/wallet'
+    import features from '@features/features'
+    import { BasicActivityDetails, Button } from 'shared/components'
+    import { ActivityRoute, activityRouter } from '../../../../../lib/routers'
+
+    export let activity: Activity
+
+    let details: Record<string, unknown>
+
+    $: activity && (details = getActivityDetails())
+
+    $: isActivityIncomingAndUnclaimed =
+        activity &&
+        activity.asyncData &&
+        (activity?.direction === ActivityDirection.Incoming || activity.isSelfTransaction) &&
+        activity.asyncData?.asyncStatus === ActivityAsyncStatus.Unclaimed
+
+    $: shouldShowActions =
+        isActivityIncomingAndUnclaimed &&
+        activity.asyncData?.asyncStatus !== ActivityAsyncStatus.Timelocked &&
+        features.dashboard.activity.actions.enabled
+
+    function getActivityDetails(): Record<string, unknown> {
+        if (!activity) {
+            return {}
+        }
+
+        const asset =
+            activity.type === ActivityType.Transaction || activity.type === ActivityType.Foundry
+                ? getAssetFromPersistedAssets(activity.assetId)
+                : undefined
+
+        const details = {
+            transactionTime: activity.time,
+            inclusionState: activity.inclusionState,
+            tag: activity.tag,
+            metadata: activity.metadata,
+            direction: activity.direction,
+            asyncStatus: activity.asyncData?.asyncStatus,
+            claimedDate: activity.asyncData?.claimedDate,
+            claimingTransactionId: activity.asyncData?.claimingTransactionId,
+            expirationDate:
+                activity.asyncData?.asyncStatus !== ActivityAsyncStatus.Claimed
+                    ? activity.asyncData?.expirationDate
+                    : null,
+            timelockDate: activity.asyncData?.timelockDate,
+            subject: activity.subject,
+        }
+        if (activity.type === ActivityType.Transaction) {
+            return {
+                ...details,
+                type: activity.type,
+                asset,
+                storageDeposit: activity.storageDeposit,
+                rawAmount: activity.rawAmount,
+                unit: asset?.metadata?.unit,
+                giftedStorageDeposit: activity.giftedStorageDeposit,
+                isInternal: activity.isInternal,
+            }
+        }
+    }
+    async function onClaim(): Promise<void> {
+        const isUnlocked = await isStrongholdUnlocked()
+        if (isUnlocked) {
+            claimActivity(activity)
+        } else {
+            $activityRouter?.goTo(ActivityRoute.Password)
```

Comment:
I think we should handle routing in the router, not here. You could add a json to the `next` function from the router and request the router directly what to do, instead of routing here. imo anything about routing should be handled inside the router or the router view, same for all appearances we have for `$activityRouter?.goTo(belula)`

Reply:
Yep you are right, I will move this to the router 👍